core/shared/src/main/scala/scalaz/zio/package.scala

```diff
@@ -2,18 +2,25 @@
 
 package scalaz
 
-import scala.AnyRef
-
 package object zio {
 
-  implicit class IONothingSyntax[A](val io: IO[Nothing, A]) extends AnyRef {
+  implicit class IOVoidSyntax[A](val io: IO[Void, A]) extends AnyRef {
     def apply[E]: IO[E, A] = io.asInstanceOf[IO[E, A]]
+    def widenError[E]: IO[E, A] = apply
+  }
+
+  implicit class IOSyntax[E, A](val io: IO[E, A]) extends AnyRef {
+    /**
+      * Widens the error type to any supertype. While `leftMap` suffices for this
+      * purpose, this method is significantly faster for this purpose.
+      */
+    def widenError[E2 >: E]: IO[E2, A] = io.asInstanceOf[IO[E2, A]]
   }
 
   type Task[A] = IO[Throwable, A]
 
-  type Unexceptional[A] = IO[Nothing, A]
+  type Unexceptional[A] = IO[Void, A]
```

Comment:
We also should probably start using `Unexceptional` (which i would like to rename to `Infallible`) wherever `IO[NoError, A]` is used.

Reply:
+1 to `Infallible` rename, although I would use it judiciously in Scalaz. For example, it could be used in the `finalizer` to `bracket` as a clear signal it may not fail.

Reply:
I've renamed `Unexceptional` to `Infallible` and replaced `IO[Void, A]` with it in several places according to John's suggestion.